contents/javascript/Javascript-framework-React.md

```diff
@@ -0,0 +1,142 @@
+<frontmatter>
+  title: React
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# React
+
+Author: [Aadyaa Maddi](https://github.com/amad-person)
+
+Reviewers: [Amrut Prabhu](https://github.com/amrut-prabhu), [Marvin Chin](https://github.com/marvinchin)
+
+### What is React?
+
+**React** is an open-source JavaScript <tooltip content="React only handles the view component of your application. It is not an MVC framework!">library</tooltip> for building composable user interfaces. It was developed by Facebook and released to the world in 2013, and is now used in a huge number of popular applications, including Facebook and Instagram.
+
+<box type="tip">
+    If you have the <a href="https://github.com/facebook/react-devtools">React DevTools</a> extension installed in your browser, you can inspect the webpage of either of these applications and observe that React is being used by them! 
+</box>
+
+React abstracts away the <tooltip content="The Document Object Model, or the 'DOM', is an interface to web pages. It represents the page as nodes and objects, allowing programs to read and manipulate the page's content, structure, and styles.">DOM</tooltip> from you, giving you a simpler programming model and better performance.
+It can also [render on the server using Node](https://alligator.io/react/server-side-rendering/), and it can power native mobile applications using [React Native](https://facebook.github.io/react-native/), which is an extension of React.
+
+### Why use React?
+
+React allows developers to create interactive web applications where data in specific parts of a page can be changed without reloading the entire page. 
+The main purpose of React is to be fast, scalable, and simple. 
+Some of its benefits are described in detail below.
+
+#### React is Declarative
+
+With React's <tooltip content="Declarative programming focuses on what the program should accomplish without specifying how the program should achieve the result.">declarative</tooltip> approach to building user interfaces, you can create interaction-rich applications. 
+You can build these interfaces without manipulating the DOM directly, and even have an event system that doesn't require you to interact with <tooltip content="Different browsers might have differences in how their DOM event implementations work. React provides its own event system so that events can work in the same way across different browsers.">actual DOM events</tooltip>. 
```

Comment:
Might need to re-sequence the explanations because it feels like claims are made, and technical terms are being introduced, without explanations. Perhaps you can start by explaining the traditional approach (DOM, imperative approach, etc.) and then show who React differs (declarative approach). Introduce technical terms when you are ready to explain them at the same time.